Common/Securities/Crypto/Crypto.cs

```diff
@@ -58,9 +59,15 @@ public Crypto(SecurityExchangeHours exchangeHours, Cash quoteCurrency, Subscript
             Holdings = new CryptoHolding(this, currencyConverter);
 
             // decompose the symbol into each currency pair
-            string baseCurrencySymbol, quoteCurrencySymbol;
-            Forex.Forex.DecomposeCurrencyPair(config.Symbol.Value, out baseCurrencySymbol, out quoteCurrencySymbol);
-            BaseCurrencySymbol = baseCurrencySymbol;
+            string quoteCurrencySymbol = symbolProperties.QuoteCurrency;
+            if (config.Symbol.Value.EndsWith(quoteCurrencySymbol))
+            {
+                BaseCurrencySymbol = config.Symbol.Value.RemoveFromEnd(quoteCurrencySymbol);
+            }
+            else
+            {
+                throw new InvalidOperationException($"symbol doesn't end with {quoteCurrencySymbol}");
+            }
```

Comment:
Maybe we could move this into its own method, similar to `Forex.DecomposeCurrencyPair` and add some unit tests for it, verifying different Crypto pair cases

Reply:
I'm okay with that. Does it mean that this new method should accept symbol and symbolProperties? The main problem having symbol props is strong restriction and usually, to my knowledge, its not available everywhere

Reply:
Agree, replied extensively in https://github.com/QuantConnect/Lean/pull/3000#issuecomment-475625604